pallets/api/src/extension.rs

```diff
@@ -292,17 +297,31 @@ mod tests {
 			}
 		}
 
-		for (version, value, expected_result) in vec![
-			(0, 10, Ok(V0(10))),
-			(0, 100, Ok(V0(50))),
-			(1, 10, Ok(V1(10))),
-			(1, 100, Ok(V1(100))),
-			(2, 10, Err(Other("DecodingFailed"))),
-		] {
-			let env = MockEnvironment { func_id: u16::from_le_bytes([0, version]), ext_id: 0u16 };
+		#[test]
+		fn versioned_result_converter_works() {
+			for (version, value, expected_result) in vec![
+				(0, 10, Ok(V0(10))),
+				(0, 100, Ok(V0(50))),
+				(1, 10, Ok(V1(10))),
+				(1, 100, Ok(V1(100))),
+			] {
+				let env =
+					MockEnvironment { func_id: u16::from_le_bytes([0, version]), ext_id: 0u16 };
+				let result = VersionedResultConverter::<u8, VersionedRuntimeResult>::try_convert(
+					value, &env,
+				);
+				assert_eq!(result, expected_result);
+			}
+		}
+
+		#[test]
+		fn versioned_result_converter_fails() {
+			let env = MockEnvironment { func_id: u16::from_le_bytes([0, 2]), ext_id: 0u16 };
 			let result =
-				VersionedResultConverter::<u8, VersionedRuntimeResult>::try_convert(value, &env);
-			assert_eq!(result, expected_result);
+				VersionedResultConverter::<u8, VersionedRuntimeResult>::try_convert(10, &env)
+					.err()
+					.unwrap(); // Again, can't use `unwrap_err()` due to missing `Debug` trait.
```

Comment:
```rs
let result = VersionedErrorConverter::<VersionedError>::convert(BadOrigin, &env).err();
assert_eq!(result, Some(Other("DecodingFailed")));
```

Refactor to get rid of the comment and `unwrap()`.

Reply:
Thank you